pbcommand/models/tool_contract.py

```diff
@@ -251,6 +251,22 @@ def __init__(self, task_id, is_distributed, input_files, output_files,
         self.nproc = nproc
         self.resources = resources
 
+    @property
+    def tmp_dir(self):
+        for resource in self.resources:
+            if "type_id" in resource:
+                if resource["type_id"] == ResourceTypes.TMP_DIR:
+                    return resource["path"]
+        return None
+
+    @property
+    def tmp_file(self):
```

Comment:
In principle, these are lists. 

Also, if it's not clear, the resource 'types' in the RTC and the TC are different.

TC:

resources = ['$tmpdir', '$tmpfile', '$tmpfile']

In the resolved TC:

```javascript
resources = [
{'type_id':'$tmpdir', 'path': "/path/to-dir"}, 
{"type_id": "$tmpfile", "path": "/path/to/t2.txt"}, 
("type_id": "$tmpfile", "path": "/path/to/t2.txt"}]
```